10311-1_1.xml

```diff
@@ -0,0 +1,194 @@
+				<Mandatory>Optional</Mandatory>
+				<Type>Time</Type>
+				<RangeEnumeration></RangeEnumeration>
+				<Units></Units>
+				<Description>The timestamp of when the measurement was performed.</Description>
+			</Item>
+			<Item ID="5700">
+				<Name>Sensor Value</Name>
+				<Operations>R</Operations>
+				<MultipleInstances>Single</MultipleInstances>
+				<Mandatory>Mandatory</Mandatory>
+				<Type>Float</Type>
+				<RangeEnumeration></RangeEnumeration>
+				<Units></Units>
+				<Description>Last or Current Measured Value from the Sensor expressed in the unit defined by the "Sensor Units" resource if present.</Description>
+			</Item>
+			<Item ID="5701">
+				<Name>Sensor Units</Name>
+				<Operations>R</Operations>
+				<MultipleInstances>Single</MultipleInstances>
+				<Mandatory>Optional</Mandatory>
+				<Type>String</Type>
+				<RangeEnumeration></RangeEnumeration>
+				<Units></Units>
+				<Description>
+                    Measurement Units Definition.
+                </Description>
+			</Item>
+			<Item ID="5750">
+				<Name>Application Type</Name>
+				<Operations>RW</Operations>
+				<MultipleInstances>Single</MultipleInstances>
+				<Mandatory>Optional</Mandatory>
+				<Type>String</Type>
+				<RangeEnumeration></RangeEnumeration>
+				<Units></Units>
+				<Description>
+                    The application type of the sensor or actuator as a string, for instance "Air Pressure".
+                </Description>
+			</Item>
+			<Item ID="5821">
+				<Name>Current Calibration</Name>
+				<Operations>RW</Operations>
+				<MultipleInstances>Single</MultipleInstances>
+				<Mandatory>Optional</Mandatory>
+				<Type>Float</Type>
+				<RangeEnumeration></RangeEnumeration>
+				<Units></Units>
+				<Description>Read or Write the current calibration coefficient</Description>
+			</Item>
+			<Item ID="6050">
+				<Name>Fractional Timestamp</Name>
+				<Operations>R</Operations>
+				<MultipleInstances>Single</MultipleInstances>
+				<Mandatory>Optional</Mandatory>
+				<Type>Float</Type>
+				<RangeEnumeration>0..1</RangeEnumeration>
+				<Units>s</Units>
+				<Description>Fractional part of the timestamp when sub-second precision is used (e.g., 0.23 for 230 ms).</Description>
+			</Item>
+			<Item ID="6042">
+				<Name>Measurement Quality Indicator</Name>
+				<Operations>R</Operations>
+				<MultipleInstances>Single</MultipleInstances>
+				<Mandatory>Optional</Mandatory>
+				<Type>Integer</Type>
+				<RangeEnumeration>0..23</RangeEnumeration>
+				<Units></Units>
+				<Description>Measurement quality indicator reported by a smart sensor. 0: UNCHECKED No quality checks were done because they do not exist or can not be applied. 1: REJECTED WITH CERTAINTY The measured value is invalid. 2: REJECTED WITH PROPABILITY The measured value is likely invalid. 3: ACCEPTED BUT SUSPICIOUS The measured value is likely OK. 4: ACCEPTED The measured value is OK. 5-15: Reserved for future expansion. 16-23: Vendor specific measurement quality.</Description>
```

Comment:
This is quite a strange thing to carry around as a measurement. I'd assume that the application logic needs to decide what to do with the measurement based on the value of 6049. I don't see why would a lwm2m client ever tell a lwm2m server what to do with the measurement, like "ACCEPTED BUT SUSPICIOUS" that is application logic, that is not a measurement.

Reply:
IIRC, this metadata about the measurement was for more advanced sensor types that have the sufficient logic to know more about the measurement quality than the LwM2M server would know (e.g., based only on 6049).

Reply:
There are multiple use cases where a weather instrument may flag a measurement with "ACCEPTED BUT SUSPICIOUS". In general, quality control checks are done in multiple levels. The first level checks are done already at the smart instrument. For example, a multi-sensor weather instrument may detect that precipitation type is snow but air temperature is +5C. This is likely a failure in precipitation type detection but can be possible in some rare cases, so it makes sense to tag the precipitation type measurement, perhaps also the temperature measurement. I'm not a meteorologist, so do not take +5C as an important threshold, but you probably get what I'm after.